extensions/amp-story-auto-ads/0.1/story-ad-page.js

```diff
@@ -0,0 +1,486 @@
+      const alternateBody = this.adDoc_.querySelector('#x-a4a-former-body');
+      alternateBody &&
+        toggleAttribute(alternateBody, PageAttributes.IFRAME_BODY_VISIBLE);
+    }
+  }
+
+  /**
+   * Create an `amp-story-page` containing an `amp-ad`.
+   * @return {!Element}
+   */
+  build() {
+    this.timeCreated_ = Date.now();
+    this.pageElement_ = this.createPageElement_();
+    this.adElement_ = this.createAdElement_();
+
+    const glassPane = this.doc_.createElement('div');
+    glassPane.classList.add(GLASS_PANE_CLASS);
+
+    const gridLayer = this.doc_.createElement('amp-story-grid-layer');
+    gridLayer.setAttribute('template', 'fill');
+
+    const paneGridLayer = gridLayer.cloneNode(/* deep */ false);
+
+    gridLayer.appendChild(this.adElement_);
+    paneGridLayer.appendChild(glassPane);
+    this.pageElement_.appendChild(gridLayer);
+    this.pageElement_.appendChild(paneGridLayer);
+
+    // Set up listener for ad-loaded event.
+    this.adElement_
+      .signals()
+      // TODO(ccordry): Investigate using a better signal waiting for video loads.
+      .whenSignal(CommonSignals.INI_LOAD)
+      .then(() => this.whenAdLoaded_());
+
+    this.analyticsEvent_(AnalyticsEvents.AD_REQUESTED, {
+      [AnalyticsVars.AD_REQUESTED]: Date.now(),
+    });
+
+    return this.pageElement_;
+  }
+
+  /**
+   * @return {!Element}
+   * @private
+   */
+  createPageElement_() {
+    const attributes = dict({
+      'ad': '',
+      'distance': '2',
+      'i-amphtml-loading': '',
+      'id': this.id_,
+    });
+
+    return createElementWithAttributes(this.doc_, 'amp-story-page', attributes);
+  }
+
+  /**
+   * @return {!Element}
+   * @private
+   */
+  createAdElement_() {
+    if (this.config_['type'] === 'fake') {
+      this.config_['id'] = `i-amphtml-demo-${this.index_}`;
+    }
+    return createElementWithAttributes(this.doc_, 'amp-ad', this.config_);
+  }
+
+  /**
+   * Things that need to happen after the created ad is "loaded".
+   * @private
+   */
+  whenAdLoaded_() {
+    // Ensures the video-manager does not follow the autoplay attribute on
+    // amp-video tags, which would play the ad in the background before it is
+    // displayed.
+    this.pageElement_.getImpl().then(impl => impl.delegateVideoAutoplay());
+
+    // Remove loading attribute once loaded so that desktop CSS will position
+    // offscren with all other pages.
+    this.pageElement_.removeAttribute(PageAttributes.LOADING);
+
+    this.analyticsEvent_(AnalyticsEvents.AD_LOADED, {
+      [AnalyticsVars.AD_LOADED]: Date.now(),
+    });
+
+    const adFrame = elementByTag(this.pageElement_, 'iframe');
+    if (adFrame) {
+      this.adDoc_ = getFrameDoc(/** @type {!HTMLIFrameElement} */ (adFrame));
+    }
+
+    this.loaded_ = true;
+
+    this.loadCallbacks_.forEach(cb => cb());
+  }
+
+  /**
+   * Try to create CTA before showing the ad. Will fail if not enough metadata
```

Comment:
Try to create CTA (Click-To-Action) button ...

Reply:
Done.